cmake/libssh2-config.cmake.in

```diff
@@ -1,6 +1,16 @@
 # Copyright (C) The libssh2 project and its contributors.
 # SPDX-License-Identifier: BSD-3-Clause
 
+set(CRYPTO_BACKEND_LIST OpenSSL wolfSSL Libgcrypt mbedTLS)
+
+if("@CRYPTO_BACKEND@" IN_LIST CRYPTO_BACKEND_LIST)
+    find_package(@CRYPTO_BACKEND@)
+endif()
+
+if(@ZLIB_FOUND@)
+    find_package(ZLIB)
```

Comment:
```suggestion
  find_package(@CRYPTO_BACKEND@)
endif()

if(@ZLIB_FOUND@)
  find_package(ZLIB)
```

Would this work with wolfSSL, which uses lowercase in the root CMakeLists.txt `find_package`? Also would it work with `WinCNG` which doesn't use `find_package`? Would these use our own find logic stored under `./cmake`? 

TL;DR: is there a solution which doesn't need duplicating dependency detection in the distributed cmake config?

Reply:
> Would this work with wolfSSL, which uses lowercase in the root CMakeLists.txt find_package? 

It will fail. I didn't notice that wolfSSL's package name and `CRYPTO_BACKEND` were different. It seems like each `CRYPTO_BACKEND` should be handled separately.

> Also would it work with WinCNG which doesn't use find_package?

WinCNG isn't included in `CRYPTO_BACKEND_LIST` for that reason.

> edit: in comparison, curl has this for OpenSSL and zlib only:
https://github.com/curl/curl/blob/39173f66e541db909069a4ce30d7590b76041596/CMake/curl-config.cmake.in#L26-L32
It's unclear why these two dependencies need this and not others, but we may go with that too if they are special. In a config file find_dependency seems to preferred over find_package?

Thanks for resource. `find_dependency()` would be more appropriate in this case. I'll try this way.

Reply:
> TL;DR: is there a solution which doesn't need duplicating dependency detection in the distributed cmake config?

I think this is the proper way to handle package dependencies in CMake.
https://cmake.org/cmake/help/latest/guide/importing-exporting/index.html#:~:text=All%20required%20dependencies,the%20Stats%20package.